pkgs/os-specific/darwin/install-binary-package/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, stdenvNoCC, makeSetupHook, _7zz, libarchive }:
+
+{ appName ? "."
+, nativeBuildInputs ? []
+, sourceRoot ? "."
+, ...
+}@args:
+
+let
+  unpackDmgPkg = makeSetupHook {
+    name = "unpack-dmg-pkg";
+    propagatedBuildInputs = [ _7zz libarchive ];
+  } ./unpack-dmg-pkg.sh;
+in
+stdenvNoCC.mkDerivation (args // {
+  dontMakeSourcesWritable = args.dontMakeSourcesWritable or true;
+  dontPatch = args.dontPatch or true;
+  dontConfigure = args.dontConfigure or true;
+  dontBuild = args.dontBuild or true;
+  dontStrip = args.dontStrip or true;
+  dontFixup = args.dontFixup or true;
+  dontCheck = args.dontCheck or true;
```

Comment:
```suggestion
```

The attribute is called `doCheck` in `stdenv.mkDerivation`, and it's `false` by default.
We don't need to touch it, unless we need to use `finalAttrs.doCheck`.

If we do need `finalAttrs.doCheck` somewhere, though, we'll still have to do
```
  doCheck = args.doCheck or false;
```

Reply:
@afh This comment slipped out of my radar. Thank you for your patience.

> For some reason I thought both existed, @ShamrockLee. Grep'ing through the packages around 80 use `dontCheck` (likely including some false positives). Is this something that could and should be cleaned up too at some point?

After some keyword searching inside VSCode, it seems most of the `dontCheck` uses refer to a [Haskell trivial helper](https://nixos.org/manual/nixpkgs/unstable/#haskell-trivial-helpers) (function) [`haskell.lib.compose.dontCheck`](https://github.com/NixOS/nixpkgs/blob/2cd70794de65b35a6e4d5e37afa3c0c2eaba097e/pkgs/development/haskell-modules/lib/compose.nix#L107-L110), which takes a derivation return one with `{ doCheck = false; }` overriding.

Searching with `dontCheck =` returns only 7 matches. Two of which is the definition of `haskell.lib.compose.dontCheck`, and the other 5 are Python packages. Since [`pkgs/development/interpreters/python/mk-python-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/2cd70794de65b35a6e4d5e37afa3c0c2eaba097e/pkgs/development/interpreters/python/mk-python-derivation.nix) doesn't provide such implementation, they are likely mistakes.

Reply:
Once again an excellent analysis from you, @ShamrockLee 😃👍

`pkgs/by-name/py/pyxel/package.nix` seems to also use `dontCheck` and is built using `buildPythonApplication` (defined in [`pkgs/development/interpreters/python/python-packages-base.nix`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/python-packages-base.nix), which also does not seem to offer `dontCheck`).
Anything we can do to confirm that the `dontCheck` used in the python packages is indeed a mistake prior to creating a PR? Or should I (unless you prefer to do it yourself) create a PR to get more eyes on this?

Reply:
ℹ️ Here are the three PRs #307823, #307835, #307838 (draft) that remove `dontCheck` from the packages mentioned earlier. Some PR also update and/or modernize some of the packages.